CoreVulkanDisplay/src/au/gov/asd/tac/constellation/visual/vulkan/CVKSwapChain.java

```diff
@@ -273,41 +278,22 @@ private int InitVKSwapChain(MemoryStack stack) {
         if (VkFailed(ret)) return ret;
 
         // Store the native handle for each image and create a view for it
-        imageHandles = new ArrayList<>(imageCount);
-        imageViewHandles = new ArrayList<>();
+        swapChainImages = new ArrayList<>(imageCount);
         imageAcquisitionHandles = new ArrayList<>();
         commandExecutionHandles = new ArrayList<>();
         renderFenceHandles = new ArrayList<>();
         for(int i = 0; i < imageCount; ++i) {
             long swapChainImageHandle = pSwapchainImageHandles.get(i);
-            imageHandles.add(swapChainImageHandle);
             
-            // Create an image view for this image
-            VkImageViewCreateInfo imageViewCreateInfo = VkImageViewCreateInfo.callocStack(stack);
-
-            imageViewCreateInfo.sType(VK_STRUCTURE_TYPE_IMAGE_VIEW_CREATE_INFO);
-            imageViewCreateInfo.image(swapChainImageHandle);
-            imageViewCreateInfo.viewType(VK_IMAGE_VIEW_TYPE_2D);
-            imageViewCreateInfo.format(cvkDevice.GetSurfaceFormat().Value());
-
-            imageViewCreateInfo.components().r(VK_COMPONENT_SWIZZLE_IDENTITY);
-            imageViewCreateInfo.components().g(VK_COMPONENT_SWIZZLE_IDENTITY);
-            imageViewCreateInfo.components().b(VK_COMPONENT_SWIZZLE_IDENTITY);
-            imageViewCreateInfo.components().a(VK_COMPONENT_SWIZZLE_IDENTITY);
-
-            imageViewCreateInfo.subresourceRange().aspectMask(VK_IMAGE_ASPECT_COLOR_BIT);
-            imageViewCreateInfo.subresourceRange().baseMipLevel(0);
-            imageViewCreateInfo.subresourceRange().levelCount(1);
-            imageViewCreateInfo.subresourceRange().baseArrayLayer(0);
-            imageViewCreateInfo.subresourceRange().layerCount(1);
-
-            LongBuffer pImageView = stack.mallocLong(1);
-            ret = vkCreateImageView(cvkDevice.GetDevice(), 
-                                    imageViewCreateInfo, 
-                                    null, //allocation callbacks
-                                    pImageView);
-            if (VkFailed(ret)) { return ret; }            
-            imageViewHandles.add(pImageView.get(0));      
+            // Factory that returns a SwapChainImage instance initialised with device and image handle
+            // And sets default paramters
+            swapChainImages.add(CVKSwapChainImage.Initialise(cvkDevice, swapChainImageHandle));
+            swapChainImages.get(i).SetExtent(cvkDevice.GetCurrentSurfaceExtent().width(),
+                    cvkDevice.GetCurrentSurfaceExtent().height());
+       
+            // Create the image view
+            swapChainImages.get(i).CreateImageView();
```

Comment:
Should this also happen in Initialise (ie a private method)

Reply:
Moved to inside Create